src/main/java/de/dennisguse/opentracks/content/provider/ContentProviderUtils.java

```diff
@@ -197,20 +200,21 @@ private void deleteTrackPointsAndWaypoints(Context context, long trackId) {
         contentResolver.delete(TrackPointsColumns.CONTENT_URI_BY_ID, where, selectionArgs);
 
         contentResolver.delete(WaypointsColumns.CONTENT_URI, WaypointsColumns.TRACKID + "=?", new String[]{Long.toString(trackId)});
-        deleteDirectoryRecurse(FileUtils.getPhotoDir(context, trackId));
     }
 
     /**
      * Delete the directory recursively.
      *
-     * @param dir the directory
+     * @param file the directory
      */
-    private void deleteDirectoryRecurse(File dir) {
-        if (dir != null && dir.exists() && dir.isDirectory()) {
-            for (File child : dir.listFiles()) {
+    private void deleteDirectoryRecurse(File file) {
+        if (file != null && file.exists() && file.isDirectory()) {
```

Comment:
Looks exactly like `cleanPhotoDir`, right?
What about moving it to FileUtils?

Reply:
I've deleted cleanPhotoDir from test but anyway I think it can be a future utility so I'm going to move it to FileUtils.